pkgs/by-name/ai/aider/package.nix

```
@@ -0,0 +1,95 @@
+{
+  lib,
+  python311,
+  fetchFromGitHub,
+  git,
+  portaudio,
+}:
+
+python311.pkgs.buildPythonApplication rec {
+  pname = "aider";
+  version = "0.48.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "paul-gauthier";
+    repo = "aider";
+    rev = "v${version}";
+    hash = "sha256-0m5ZHCfxlOOeUvfQznF5hTCJANCBtrO9rWDudQ+RUxM=";
+  };
+
+  build-system = with python311.pkgs; [
+    pythonRelaxDepsHook
+    setuptools
+  ];
+
+  dependencies = with python311.pkgs; [
+    aiohappyeyeballs
+    backoff
+    beautifulsoup4
+    configargparse
+    diff-match-patch
+    diskcache
+    flake8
+    gitpython
+    grep-ast
+    importlib-resources
+    jsonschema
+    llama-index-core
+    llama-index-embeddings-huggingface
+    litellm
+    networkx
+    numpy
+    packaging
+    pathspec
+    pillow
+    playwright
+    prompt-toolkit
+    pypandoc
+    pyyaml
+    rich
+    scipy
+    sounddevice
+    soundfile
+    streamlit
+    watchdog
+  ];
```

This worked for me for darwin:
```suggestion
  dependencies = with python311.pkgs; ([
    aiohappyeyeballs
    backoff
    beautifulsoup4
    configargparse
    diff-match-patch
    diskcache
    flake8
    gitpython
    grep-ast
    importlib-resources
    jsonschema
    litellm
    networkx
    numpy
    packaging
    pathspec
    pillow
    playwright
    prompt-toolkit
    pypandoc
    pyyaml
    rich
    scipy
    sounddevice
    soundfile
    streamlit
    watchdog
  ] ++ lib.optionals (!tensorflow.meta.broken) [
    llama-index-core
    llama-index-embeddings-huggingface
  ]);
```

---

the parens around the with expr are unnecessary (and would probably fail the CI?)

---

They are necessary. Otherwise we'd have to repeat the `with ...` after the optionals check.

---

TIL!